src/integrationTest/java/uk/gov/hmcts/reform/preapi/tasks/AddNROUsersIT.java

```diff
@@ -0,0 +1,347 @@
+package uk.gov.hmcts.reform.preapi.tasks;
+
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.preapi.entities.AppAccess;
+import uk.gov.hmcts.reform.preapi.repositories.AppAccessRepository;
+import uk.gov.hmcts.reform.preapi.repositories.CourtRepository;
+import uk.gov.hmcts.reform.preapi.repositories.RoleRepository;
+import uk.gov.hmcts.reform.preapi.repositories.UserRepository;
+import uk.gov.hmcts.reform.preapi.security.service.UserAuthenticationService;
+import uk.gov.hmcts.reform.preapi.services.UserService;
+import uk.gov.hmcts.reform.preapi.util.HelperFactory;
+import uk.gov.hmcts.reform.preapi.utils.IntegrationTestBase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Slf4j
+public class AddNROUsersIT extends IntegrationTestBase {
+
+    @Autowired
+    private AppAccessRepository appAccessRepository;
+    @Autowired
+    private CourtRepository courtRepository;
+    @Autowired
+    private RoleRepository roleRepository;
+    @Autowired
+    private static UserAuthenticationService userAuthenticationService;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private UserService userService;
+
+    private static final String TEST_USERS_FILE =
+        "src/integrationTest/java/uk/gov/hmcts/reform/preapi/utils/Test_NRO_User_Import.csv";
+    private static final String CRON_USER_EMAIL = "Phoebe.Revolta@HMCTS.net";
+    private ArrayList<ImportedNROUser> testImportedNROUsers;
+    private ArrayList<String> failureCaseEmails;
+
```

Comment:
Probably a good idea to change that email. Also, generally better practice to use `List<>` instead of `ArrayList<>` unless you are using `ArrayList<>` specific methods
```suggestion
    private static final String CRON_USER_EMAIL = "example@example.com";
    private List<ImportedNROUser> testImportedNROUsers;
    private List<String> failureCaseEmails;
```